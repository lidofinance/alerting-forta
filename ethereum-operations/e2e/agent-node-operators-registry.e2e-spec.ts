import { configureContainer, Finding } from "forta-agent";
import { AwilixContainer, asFunction } from "awilix";
import {
  provideAgentPath,
  provideRunBlock,
  provideRunTransaction,
} from "./utils";

const TEST_TIMEOUT = 60_000; // ms

describe("agent-node-operators-registry e2e tests", () => {
  let runBlock: (blockHashOrNumber: string | number) => Promise<Finding[]>;
  let runTransaction: (txHash: string) => Promise<Finding[]>;
  let logSpy: jest.SpyInstance;

  beforeAll(() => {
    logSpy = jest.spyOn(console, "log");
    logSpy.mockImplementation(() => {});
  });

  beforeEach(async () => {
    const container = configureContainer() as AwilixContainer;
    container.register({
      agentPath: asFunction(
        provideAgentPath(
          "subagents/node-operators-registry/agent-node-operators-registry",
        ),
      ),
      runTransaction: asFunction(provideRunTransaction),
      runBlock: asFunction(provideRunBlock),
    });

    // https://docs.forta.network/en/latest/cli/#invoke-commands-programmatically
    runTransaction = container.resolve("runTransaction");
    runBlock = container.resolve("runBlock");
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it(
    "should process tx with removed signing keys",
    async () => {
      // todo: should be uncommented after v2
      // const findings = await runTransaction(
      //   "0x0397a4af942b58698cce7b5e7610dc7da41309b311318a5bc5ff54356bca74e7"
      // );
      // expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should process tx with set NO Stake limit by NON-EasyTrack action",
    async () => {
      // todo: should be uncommented after v2
      // const findings = await runTransaction(
      //   "0x06c62f1d1fcfe2cd92c9b1e571b7bcc37d9ee541a6fae7be48d3f5bae2e07d3c"
      // );
      // expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should process tx with added Node operator",
    async () => {
      // todo: should be uncommented after v2
      // const findings = await runTransaction(
      //   "0x05ffd9394c9a28f361b7ff1b135a52c173fe5be55e3e639cf7088f0436903aab"
      // );
      // expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );
});
