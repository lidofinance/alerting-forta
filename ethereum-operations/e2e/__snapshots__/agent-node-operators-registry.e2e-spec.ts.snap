// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agent-node-operators-registry e2e tests should process tx with added Node operator 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "NODE-OPERATOR-ADDED",
  "description": "Node operator 29 added
Name: Attestant (BVI) Limited
Reward address: 0xcd51df5142156BD92E7A9494Dd2b9362f5078D64
StakingLimit: 0",
  "labels": Array [],
  "metadata": Object {
    "args": "29,Attestant (BVI) Limited,0xcd51df5142156BD92E7A9494Dd2b9362f5078D64,0",
  },
  "name": "â„¹ NO Registry: Node operator added",
  "protocol": "ethereum",
  "severity": 1,
  "type": 4,
}
`;

exports[`agent-node-operators-registry e2e tests should process tx with removed signing keys 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "NODE-OPERATORS-KEYS-REMOVED",
  "description": "1 signing keys removed",
  "labels": Array [],
  "metadata": Object {},
  "name": "ðŸš¨ NO Registry: Signing keys removed",
  "protocol": "ethereum",
  "severity": 4,
  "type": 4,
}
`;

exports[`agent-node-operators-registry e2e tests should process tx with set NO Stake limit by NON-EasyTrack action 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "NODE-OPERATORS-STAKING-LIMIT-SET",
  "description": "Staking limit for node operator 13 was set to 3800 by NON-EasyTrack motion!",
  "labels": Array [],
  "metadata": Object {},
  "name": "ðŸš¨ NO Stake limit set by NON-EasyTrack action",
  "protocol": "ethereum",
  "severity": 4,
  "type": 4,
}
`;
